src/utils/mtev_lru.c

```diff
@@ -0,0 +1,252 @@
+#include "mtev_lru.h"
+#include "mtev_hash.h"
+#include "mtev_log.h"
+
+#include <ck_hs.h>
+#include <sys/queue.h>
+
+struct lru_entry {
+  void *entry;
+  size_t key_len;
+  TAILQ_ENTRY(lru_entry) list_entry;
+  char key[];
+};
+
+struct mtev_lru {
+  ck_hs_t hash;
+  int32_t max_entries;
+  TAILQ_HEAD(lru_list, lru_entry) lru_cache;
+  int32_t lru_cache_size;
+  uint32_t expire_count;
+  void (*free_fn)(void *);
+  pthread_mutex_t mutex;
+};
+
+static void *
+lru_malloc(size_t r)
+{
+  return malloc(r);
+}
+
+static void
+lru_free(void *p, size_t b, bool r)
+{
+  (void)b;
+  (void)r;
+  free(p);
+  return;
+}
+
+static struct ck_malloc malloc_ck_hs = {
+  .malloc = lru_malloc,
+  .free = lru_free
+};
+
+static int rand_init = 0;
+
+#define HS_HF_STRING(F)                         \
+  static unsigned long                          \
+  F(const void *k, unsigned long seed)          \
+  {                                             \
+    return mtev_hash__hash(k, strlen(k), seed);   \
+  }
+
+HS_HF_STRING(lru_entry_hash);
+
+#define offset_of(type, member)                 \
+  ((size_t)(&((type *)0)->member))
+
+#define container_of(derived_ptr, type, field)                \
+  ((type *)((char *)(derived_ptr) - offset_of(type, field)))
+
+static void
+hs_init(ck_hs_t *hs, unsigned int mode, ck_hs_hash_cb_t *hf, ck_hs_compare_cb_t *cf, unsigned long size)
+{
+  if(!rand_init) {
+    srand48((long int)time(NULL));
```

Comment:
OK... then we should have this code use the same rand-init that `mtev_hash_init` does, maybe extract that to another function that this can call? I don't really understand how libck uses that random number, but it makes it feel a little worse that this will make `mtev_hash_init` and `mtev_lru` will probably get the same output from `lrand48`...